src/base/AsyncJobCalls.h

```diff
@@ -11,6 +11,7 @@
 
 #include "base/AsyncJob.h"
 #include "base/CbcPointer.h"
+#include "base/IoManip.h"
```

Comment:
We should not force manipulators on all AsyncJobCalls users because most calls do not use these manipulators. In general, IoManip.h should be included in .cc files that use/need it, not header files (that do not).

Reply:
The problem for this is the code in the header file using the std::optional printer. Whenever those templates **attempt** to expand the error appears.

Reply:
> The problem for this is the code in the header file using the std::optional printer.

Some code that uses these templates instantiates them with std::optional. That code should worry about printing the arguments it instantiates. Such code is (currently) quite rare.

These templates and their many other users (i.e. the ones that do not instantiate the call templates with std::optional) should not be exposed to std::optional. std::optional is just one of the numerous possible call parameter types. We do not want to `#include` all of those parameter-declaring headers here. It is harmful and should not be necessary.

Thank you for posting the logs. I will try to figure out what is going on.

Reply:
_Status update_: I believe I figured out why GCC refuses to print `std::optional<Ip::Address>` in icap/Xaction.cc context and found a solution that does not require exposing every async call user to std::optional printing (and such -- the exposure list can only grow if we start going down this path). I will post an update with details after investigating the SwapMetaView problem.

Reply:
>  I will post an update with details after investigating the SwapMetaView problem.

Done at https://github.com/squid-cache/squid/pull/1213#issuecomment-1360863597.